Neos.ContentRepository.Core/Classes/SharedModel/Node/NodeIdentity.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\SharedModel\Node;
+
+use Neos\ContentRepository\Core\DimensionSpace\DimensionSpacePoint;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+
+/**
+ * The content-repository-id content-stream-id dimension-space-point and the node-aggregate-id
+ * Are used in combination to distinctly identify a single node.
+ *
+ * By using getting the content graph for the content-repository-id,
+ * one can build a subgraph with the perspective to find this node:
+ *
+ *      $subgraph = $contentgraph->getSubgraph(
+ *          $nodeIdentity->contentStreamId,
+ *          $nodeIdentity->dimensionSpacePoint,
+ *          // show all disabled nodes
+ *          VisibilityConstraints::withoutRestrictions()
+ *      );
+ *      $node = $subgraph->findNodeById($nodeIdentity->nodeAggregateId);
+ *
+ * @api
+ */
+final readonly class NodeIdentity implements \JsonSerializable
+{
+    public function __construct(
+        public ContentRepositoryId $contentRepositoryId,
+        public ContentStreamId $contentStreamId,
```

Comment:
Mmm, good question, I guess both have their merit, because eg. in a future conflict resolution  thing I could see need to address a contentStream that is "parked" to the side because it had conflicts and is not currently exposed via a workspace? IDK though, we could also just decide that whenever we need access to such a content stream it gets a workspace assigned?

Reply:
or we make this a union type? or make two node identities? (which feels weird)